### ISSUE TITLE
Fatal SyntaxError: Unexpected token _ in JSON at position 176

### ISSUE THREAD

**Issue by smendola** [shiwano/typhen#10]
Thanks for addressing #9!
Now I'm trying typhen on a real-life source file, not just a hello-world.d.ts.
And I'm getting a different failure.

```
% cat sds.d.ts 

export namespace SDS {
    /**
     * A short string with no special characters
     * safe to use as an identifier in many contexts.
     */
    type SafeString = string;

    /**
     * The generic param T here does nothing, but it helps document intent
     * e.g. ElementId<QuestionElement> is more meaningful than just ElementId
     */
    type ElementId<T extends Element> = string;

    /**
     * An SDS API Element
     */
    interface Element {
        _id?: ElementId<any>,
        elementType: SafeString
    }

    /**
     * All translatable strings are represented like this
     */
    interface I18NString {
        key?: SafeString,
        specText: string
    }

    /**
     *  SW stores scalar values in an IG/IT.
     *  Each IT may a "Short Text" for use in reports.
     */
    interface SingleStorage {
        dataType: DataTypes.ScalarDataType,
        IG?: SafeString,
        IT?: SafeString,
        shortText?: I18NString
    }

    /**
     * For questions that output multiple answers, SW stores those
     * answers in individual IT's.  So each possible answer needs
     * its own IT, and therefore, its own shortText.
     */
    interface MultiStorage {
        dataType: DataTypes.VectorDataType,
        IG: SafeString,
        ITMap: {
            [code: number]: {
                IT: SafeString,
                shortText?: I18NString
            }
        }
    }

    namespace Elements {

        type QuestionOrGroup = (Question | QuestionGroup);

        interface QuestionBase extends Element {
            presentation: {
                questionText: I18NString,
                widget?: SafeString,
                widgetConfig?: object // we have no control over this
            }
            storage?: SingleStorage | MultiStorage,
        }

        interface Question extends QuestionBase {
            elementType: 'Question',
        }

        interface QuestionGroup extends QuestionBase {
            elementType: 'QuestionGroup',
            subQuestions?: ElementId<Question>[];
        }

        interface CodeList extends Element {
            elementType: 'CodeList',
            name?: SafeString,
            codes: {
                // e.g. {0:{key:'GOOD', specText:'Good'}, 1:{key:'BAD', specText:'Lousy'}}
                [code: number]: I18NString
            }
        }
    }

    /**
     */
    namespace DataTypes {
        type DataType = ScalarDataType | VectorDataType;

        type ScalarDataType =
            CodedChoice |
            FixedPoint | FloatingPoint | Integer |
            Boolean | Text | InkVector |
            LocalDate | LocalTime | LocalDateTime | Instant;

        type VectorDataType = CodedMultipleChoice;

        interface FixedPoint {
            baseType: 'FixedPoint',
            min?: number,
            max?: number,
            numDecimals?: number
        }

        interface Integer {
            baseType: 'Integer',
            min?: number,
            max?: number
        }

        interface FloatingPoint {
            baseType: 'FloatingPoint',
            min?: number,
            max?: number
        }

        interface Boolean {
            baseType: 'Boolean'
        }

        interface Text {
            baseType: 'Boolean',
            multiLine: boolean,
            maxLength?: number
        }

        interface InkVector {
            baseType: 'InkVector'
        }

        interface Instant {
            baseType: 'Instant',
        }

        interface LocalDate {
            baseType: 'LocalDate',
        }

        interface LocalTime {
            baseType: 'LocalTime',
        }

        interface LocalDateTime {
            baseType: 'LocalDateTime',
        }

        interface CodedChoice {
            baseType: 'CodedChoice',
            codeList: ElementId<Elements.CodeList>;
        }

        interface CodedMultipleChoice {
            baseType: 'CodedMultipleChoice',
            codeList: ElementId<Elements.CodeList>;
        }
    }
}

% tsc sds.d.ts

% sudo npm -g up typhen

% typhen --version
v1.2.8

% sudo npm -g up typhen-json-schema              

% typhen -V -p typhen-json-schema -d . sds.d.ts
Parsing TypeScript files
[13:23:56][DEBUG] Loading the TypeScript files
[13:23:56][DEBUG] Reading: /home/horizon/test-typhen/sds.d.ts
[13:23:56][DEBUG] Reading dafaultLibFile data
[13:23:57][DEBUG] Compiling the TypeScript files
[13:23:57][DEBUG] Parsing the TypeScript symbols
[13:23:57][DEBUG] Creating Module: module=, name="/home/horizon/test-typhen/sds", declarations=sds.d.ts(2,0)
[13:23:57][DEBUG] Creating Module: module=sds, name=SDS, declarations=sds.d.ts(2,0)
[13:23:57][DEBUG] Creating Module: module=sds.SDS, name=Elements, declarations=sds.d.ts(58,4)
[13:23:57][DEBUG] Creating UnionType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.Elements, name=Question, declarations=sds.d.ts(71,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=elementType, declarations=sds.d.ts(72,12)
[13:23:57][DEBUG] Creating ObjectType: module=sds.SDS.Elements, name=, declarations=sds.d.ts(63,26)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS, name=I18NString, declarations=sds.d.ts(26,4)
[13:23:57][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=key, declarations=sds.d.ts(27,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=specText, declarations=sds.d.ts(28,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=questionText, declarations=sds.d.ts(64,16)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=widget, declarations=sds.d.ts(65,16)
[13:23:57][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=widgetConfig, declarations=sds.d.ts(66,16)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=presentation, declarations=sds.d.ts(63,12)
[13:23:57][DEBUG] Creating UnionType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Interface: module=sds.SDS, name=SingleStorage, declarations=sds.d.ts(35,4)
[13:23:57][DEBUG] Creating UnionType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Module: module=sds.SDS, name=DataTypes, declarations=sds.d.ts(93,4)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=CodedChoice, declarations=sds.d.ts(153,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(154,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=codeList, declarations=sds.d.ts(155,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=FixedPoint, declarations=sds.d.ts(104,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(105,12)
[13:23:57][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=min, declarations=sds.d.ts(106,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=max, declarations=sds.d.ts(107,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=numDecimals, declarations=sds.d.ts(108,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=FloatingPoint, declarations=sds.d.ts(117,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(118,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=min, declarations=sds.d.ts(119,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=max, declarations=sds.d.ts(120,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=Integer, declarations=sds.d.ts(111,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(112,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=min, declarations=sds.d.ts(113,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=max, declarations=sds.d.ts(114,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=Boolean, declarations=sds.d.ts(123,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(124,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=Text, declarations=sds.d.ts(127,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(128,12)
[13:23:57][DEBUG] Creating PrimitiveType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=multiLine, declarations=sds.d.ts(129,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=maxLength, declarations=sds.d.ts(130,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=InkVector, declarations=sds.d.ts(133,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(134,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=LocalDate, declarations=sds.d.ts(141,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(142,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=LocalTime, declarations=sds.d.ts(145,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(146,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=LocalDateTime, declarations=sds.d.ts(149,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(150,12)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=Instant, declarations=sds.d.ts(137,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(138,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=dataType, declarations=sds.d.ts(36,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=IG, declarations=sds.d.ts(37,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=IT, declarations=sds.d.ts(38,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=shortText, declarations=sds.d.ts(39,8)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS, name=MultiStorage, declarations=sds.d.ts(47,4)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.DataTypes, name=CodedMultipleChoice, declarations=sds.d.ts(158,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=baseType, declarations=sds.d.ts(159,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.DataTypes, name=codeList, declarations=sds.d.ts(160,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=dataType, declarations=sds.d.ts(48,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=IG, declarations=sds.d.ts(49,8)
[13:23:57][DEBUG] Creating ObjectType: module=sds.SDS, name=, declarations=sds.d.ts(50,15)
[13:23:57][DEBUG] Creating ObjectType: module=sds.SDS, name=, declarations=sds.d.ts(51,28)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=IT, declarations=sds.d.ts(52,16)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=shortText, declarations=sds.d.ts(53,16)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=ITMap, declarations=sds.d.ts(50,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=storage, declarations=sds.d.ts(68,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=_id, declarations=sds.d.ts(19,8)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.Elements, name=QuestionBase, declarations=sds.d.ts(62,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=presentation, declarations=sds.d.ts(63,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=storage, declarations=sds.d.ts(68,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=_id, declarations=sds.d.ts(19,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=elementType, declarations=sds.d.ts(20,8)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS, name=Element, declarations=sds.d.ts(18,4)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=_id, declarations=sds.d.ts(19,8)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=elementType, declarations=sds.d.ts(20,8)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.Elements, name=QuestionGroup, declarations=sds.d.ts(75,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=elementType, declarations=sds.d.ts(76,12)
[13:23:57][DEBUG] Creating Array: module=, name=Array, declarations=../../../usr/lib/node_modules/typhen/node_modules/typescript/lib/@@__typhen.lib.d.ts(1132,0),../../../usr/lib/node_modules/typhen/node_modules/typescript/lib/@@__typhen.lib.d.ts(1308,14)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=subQuestions, declarations=sds.d.ts(77,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=presentation, declarations=sds.d.ts(63,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=storage, declarations=sds.d.ts(68,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=_id, declarations=sds.d.ts(19,8)
[13:23:57][DEBUG] Creating Interface: module=sds.SDS.Elements, name=CodeList, declarations=sds.d.ts(80,8)
[13:23:57][DEBUG] Creating StringLiteralType: module=, name=, declarations=
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=elementType, declarations=sds.d.ts(81,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=name, declarations=sds.d.ts(82,12)
[13:23:57][DEBUG] Creating ObjectType: module=sds.SDS.Elements, name=, declarations=sds.d.ts(83,19)
[13:23:57][DEBUG] Creating Property: module=sds.SDS.Elements, name=codes, declarations=sds.d.ts(83,12)
[13:23:57][DEBUG] Creating Property: module=sds.SDS, name=_id, declarations=sds.d.ts(19,8)
[13:23:57][DEBUG] Creating TypeAlias: module=sds.SDS.Elements, name=QuestionOrGroup, declarations=sds.d.ts(60,8)
[13:23:57][DEBUG] Creating UnionType: module=, name=, declarations=
[13:23:57][DEBUG] Creating TypeAlias: module=sds.SDS.DataTypes, name=DataType, declarations=sds.d.ts(94,8)
[13:23:57][DEBUG] Creating TypeAlias: module=sds.SDS.DataTypes, name=ScalarDataType, declarations=sds.d.ts(96,8)
[13:23:57][DEBUG] Creating TypeAlias: module=sds.SDS.DataTypes, name=VectorDataType, declarations=sds.d.ts(102,8)
[13:23:57][DEBUG] Creating TypeAlias: module=sds.SDS, name=SafeString, declarations=sds.d.ts(7,4)
[13:23:57][DEBUG] Creating TypeAlias: module=sds.SDS, name=ElementId, declarations=sds.d.ts(13,4)
[13:23:57][DEBUG] Validating the typhen symbols
[13:23:57] Parsed sds.d.ts
Generating files
[13:23:57][DEBUG] Rendering: templates/base.hbs, DataType
[13:23:57][DEBUG] Reading: /usr/lib/node_modules/typhen-json-schema/templates/base.hbs
[13:23:57][DEBUG] Compiling the Template: templates/base.hbs
[13:23:57][DEBUG] Rendering: templates/base.hbs, ScalarDataType
[13:23:57][DEBUG] Rendering: templates/base.hbs, QuestionOrGroup
[13:23:57][DEBUG] Rendering: templates/base.hbs, Boolean
Fatal SyntaxError: Unexpected token _ in JSON at position 176
    at JSON.parse (<anonymous>)
    at formatResult (/usr/lib/node_modules/typhen-json-schema/index.js:49:52)
    at /usr/lib/node_modules/typhen-json-schema/index.js:131:13
    at Array.forEach (native)
    at Plugin.generate (/usr/lib/node_modules/typhen-json-schema/index.js:127:13)
    at /usr/lib/node_modules/typhen/lib/runner.js:40:54
    at Promise._execute (/usr/lib/node_modules/typhen/node_modules/bluebird/js/release/debuggability.js:300:9)
    at Promise._resolveFromExecutor (/usr/lib/node_modules/typhen/node_modules/bluebird/js/release/promise.js:483:18)
    at new Promise (/usr/lib/node_modules/typhen/node_modules/bluebird/js/release/promise.js:79:10)
    at Runner.run (/usr/lib/node_modules/typhen/lib/runner.js:29:16)
    at Object.run (/usr/lib/node_modules/typhen/lib/index.js:19:49)
    at Object.<anonymous> (/usr/lib/node_modules/typhen/bin/typhen:86:12)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)

[2]

% 

```

**Comment by smendola** [comment]
This issue may actually be with typhen-json-schema, not typhen itself.

**Comment by shiwano** [comment]
Sorry for my slow response :disappointed_relieved:   I will fix it soon.

I think too that the issue is with `typhen-json-schema`.

**Comment by shiwano** [comment]
I've fixed the issue with typhen-json-schema v0.5.0.

Thank you for your reporting!

**Comment by smendola** [comment]
Thank you!
